src/AppWithRouter.js

```diff
@@ -21,37 +21,66 @@ import { createRouters } from './pages/routers';
 import { getDict } from './locales';
 import zhCN from 'rsuite/lib/IntlProvider/locales/zh_CN';
 import enUS from 'rsuite/lib/IntlProvider/locales/en_US';
-
-import { DirectionContext } from '@/components/Context';
+import loadCssFile from '@/utils/loadCssFile';
+import { readTheme, writeTheme, getThemeCssPath, getThemeId } from '@/utils/themeHelpers';
+import { ThemeContext } from '@/components/Context';
 
 const html = document.querySelector('html');
 
 export default locale => {
   class AppRouters extends React.Component {
     constructor(props) {
       super(props);
-      const direction = localStorage.getItem('direction') || 'ltr';
+      const [theme, direction] = readTheme();
       this.state = {
-        direction
+        direction,
+        theme
       };
-      html.dir = direction;
+      //skip loading default ltr theme
+      if (theme != 'default' || direction != 'ltr')
+        this.loadTheme(theme, direction);
     }
 
+    loadTheme = (themeName, direction) => {
+      writeTheme(themeName,direction);
+      const { globalVars = {} } = window.less || {};
+      window.less &&
+        window.less.modifyVars({
+          ...globalVars,
+          '@theme-is-default': themeName === 'default'
+        });
+      html.dir = direction;
+      const themeId = getThemeId(themeName, direction);
+      loadCssFile(getThemeCssPath(themeName, direction),themeId).then(() => {
```

Comment:
When change style file , the page will shaking like this.

![QQ20191129-103226-HD](https://user-images.githubusercontent.com/15609339/69839719-b25aa980-1293-11ea-9ae0-9193839455e2.gif)

How about added a loader in page when loading css do you think?